encrypt/ecies/ecies_test.go

```diff
@@ -44,3 +48,36 @@ func TestECIESFailCiphertext(t *testing.T) {
 	_, err = Decrypt(suite, private, ciphertext, nil)
 	require.NotNil(t, err)
 }
+
+func BenchmarkECIES(b *testing.B) {
+	suites := []struct {
+		kyber.Group
+	}{
+		{edwards25519.NewBlakeSHA256Ed25519()},
+		{curve25519.NewBlakeSHA256Curve25519(false)},
+		{curve25519.NewBlakeSHA256Curve25519(true)},
+		{nist.NewBlakeSHA256P256()},
+		{nist.NewBlakeSHA256QR512()},
+	}
+
+	message := make([]byte, 100_000)
+	_, _ = rand.Read(message)
+	rand := random.New()
+
+	for _, suite := range suites {
+		private := suite.Scalar().Pick(rand)
+		public := suite.Point().Mul(private, nil)
+
+		for i := 0; i < b.N; i++ {
+			var ct []byte
+			b.Run("Encrypt/"+suite.String(), func(b *testing.B) {
+				ct, _ = Encrypt(suite, public, message, nil)
+
+			})
+
+			b.Run("Decrypt/"+suite.String(), func(b *testing.B) {
+				_, _ = Decrypt(suite, private, ct, nil)
+			})
+		}
```

Comment:
This looks like the for loops should be in the sub-benchmarks, rather than in the outer one.

Reply:
Fixed !